### DISCUSSION TITLE
Tea errors w. `panic:invalid version: undefined` if there is an `action.yml` file present

### DISCUSSION THREAD

**Discussion by gaggle** [pkgxdev/pkgx#693] [Help] [answered]
Hey all,

I created this issue last week: https://github.com/teaxyz/cli/issues/690, and took a stab at fixing it in this PR https://github.com/teaxyz/cli/pull/691. Maybe I should've created a discussion here to get focus on it though?

The problem is Tea panics:

```sh
$ tea -SE
pantries sync’d ⎷

panic: spilt tea. we’re sorry and we’ll fix it… but you have to report the bug!

    https://github.com/teaxyz/cli/issues/new?title=panic%3Ainvalid+version%3A+undefined

----------------------------------------------------->> attachment begin
Error: invalid version: undefined
    at https://deno.land/x/libtea@v0.10.7/src/utils/semver.ts:44:27
    at Array.flatMap (<anonymous>)
    at new SemVer (https://deno.land/x/libtea@v0.10.7/src/utils/semver.ts:25:37)
    at https://deno.land/x/libtea@v0.10.7/src/utils/semver.ts:139:34
    at Array.map (<anonymous>)
    at new Range (https://deno.land/x/libtea@v0.10.7/src/utils/semver.ts:121:56)
    at supp (file:///private/var/folders/8b/n_gmhmk57ks85swq3mrdbrwc0000gp/T/42af7169/src/hooks/useVirtualEnv.ts:219:29)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async default (file:///private/var/folders/8b/n_gmhmk57ks85swq3mrdbrwc0000gp/T/42af7169/src/hooks/useVirtualEnv.ts:52:17)
    at async run (file:///private/var/folders/8b/n_gmhmk57ks85swq3mrdbrwc0000gp/T/42af7169/src/app.main.ts:43:30)
<<------------------------------------------------------- attachment end
```

Due to the presence of `action.yml` file:

```sh
tea -vvv -SE
pantries sync’d ⎷
{
  tea: ".../.tea/tea.xyz/v0.39.6/bin/tea",
  deno: "1.36.0",
  v8: "11.6.189.12",
  typescript: "5.1.6"
}
.../action.yml
error: {
  "cause": {
    "string": ".../action.yml"
  }
}
```

When I move `action.yml` away the problem goes away. 

https://github.com/teaxyz/cli/blob/080e540afe4c338dc7d07c55a25e6c40c61f9acd/src/hooks/useVirtualEnv.ts#L206-L213 

It looks like #L211 assumes there is always a version available?

**Comment by jhheider** [comment, marked as answer]
yes, the fix is almost certainly testing if `v` is semver, before the pkgs.push line.

**Reply by gaggle** [reply]
Okay cool, yeah, is there anything I can do to move along a fix? Happy to rework the PR if it needs a different direction or similar. I'm currently blocked until this lands and gets released.

(edit: also fine if there isn't!, appreciate y'all's hard work, just asking in case there is something)

**Reply by mxcl** [reply]
Was the node just empty? We should have fixed that in v1-alpha.2
